### ISSUE TITLE
I want to disable offset

### ISSUE THREAD

**Issue by alehosaini** [yajra/laravel-datatables#1332]
Hi @yajra, my table has more than 1,000,000 records so when i get data it takes more than 45 second to come out then i figure out the offset in eloquent is the reason.
So is there any parameter to disable offset function so i can get data in a special way. -->
Below code is explain more, i want my pagination way like this.

```			$lastId = ($request->last_id) ? $request->last_id : 0;
			$Contacts = $Contacts->where(DB::raw('contacts.id'),'>',$lastId)->limit(20);
```
- Operating System Windows
- PHP Version 7
- Laravel Version 5.2
- Laravel-Datatables Version 7

**Comment by yajra** [comment]
@alehosaini just make sure you add `->get()` on your query. Limit is automatically set by the package which comes the js ajax request.

**Comment by alehosaini** [comment]
Yes mr @yajra you right,i actually was know your solution but my problem was with performance in other word the query is very slow. you can take look to my code below:

the code work fine with data less than 100K but i was trying to get data more than 1M,
then the query took 45s to come.
then i figure out the offset function in mysql query was the reason.
even i used `->setTotalRecords($countAll)` to fix part of the problem

**Some sources:**
https://stackoverflow.com/questions/4481388/why-does-mysql-higher-limit-offset-slow-the-query-down
https://stackoverflow.com/questions/8467104/limit-with-offset-limit-500000-10-is-slow-even-after-index?noredirect=1&lq=1

Maybe you can fix this problem by trying to put parameter like **setTotalRecords**.

```php $select = [
			DB::raw('contacts.id as id'),
			DB::raw('contacts.level'),
			DB::raw('contacts.firstname'),
			DB::raw('contacts.lastname'),
			DB::raw('CONCAT(contacts.firstname," ",contacts.lastname) as contact_name'),
			DB::raw('(SELECT value FROM book WHERE module = "Contacts" AND type = "Number" AND module_id = contacts.id LIMIT 1) as tel'),
			DB::raw('(SELECT value FROM book WHERE module = "Contacts" AND type = "Email" AND module_id = contacts.id LIMIT 1) as email'),
			DB::raw('(SELECT date FROM activities WHERE type != "Note" AND contact_id = contacts.id ORDER BY id DESC LIMIT 1) as last_relation'),
			DB::raw('company.name as company'),
			DB::raw('company.id as company_id'),
			DB::raw('owner.name as owner_name')
			];
			$Contacts = Contact::leftJoin('users as owner', function($join) {
				$join->on('contacts.owner_id', '=', 'owner.id');
				})->leftJoin('companies as company', function($join) {
				$join->on('contacts.company_id', '=', 'company.id');
				})->leftJoin('activities as activity', function($join) {
				$join->on('contacts.id', '=', 'activity.contact_id')->where('activity.type','!=','Note');
			})->groupBy('contacts.id');
			$Contacts = $Contacts->select($select)->where('contacts.client_id',auth()->user()->client_id);
$countAll = Contact::where('client_id',auth()->user()->client_id)->count();
			return Datatables::eloquent($Contacts)->addColumn('DT_RowId','{{ strtolower($id) }}')->filterColumn('contact_name', function($q, $k) {
				$q->where(DB::raw("CONCAT(contacts.firstname,' ',contacts.lastname)"),"LIKE","%$k%")->orWhere("contacts.code","LIKE","%$k%");
				})->filterColumn('company', function($q, $k) {
				$q->whereRaw("company.name LIKE ?", ["%{$k}%"]);
				})->filterColumn('email', function($q, $k) {
				$q->whereRaw("contacts.email LIKE ?", ["%{$k}%"]);
				})->filterColumn('owner_name', function($q, $k) {
				$q->whereRaw("owner.name LIKE ?", ["%{$k}%"]);
			})->setTotalRecords($countAll)->make(true);

**Comment by yajra** [comment]
@alehosaini thanks for the reference. I think I got what you are trying to achieve here. Tried playing with the solutions on the given links. Will send a PR on how I see this would work and kindly test.

```php
        return Datatables::eloquent(User::query())
                         ->limit(function ($query) {
                             $query->where('id', '>', request('start'));
                         })
                         ->setTotalRecords(100)
                         ->make(true);
```

**Comment by yajra** [comment]
PR #1338 submitted. Kindly review and advise. Thanks!

**Comment by alehosaini** [comment]
Hi mr @yajra, Thanks for your updates.
I tried to work with changes but actually i have not a background with github pull requests feature.
but which i did is copying QueryBuilderEngine.php file content to my workspace in this path
`vendor\yajra\laravel-datatables-oracle\src\Engines`

then i changed part of my code with this
```php return Datatables::eloquent($Contacts)->limit(function ($query) {
            $query->where('id', '>', request('start'));
        })->addColumn('DT_RowId','{{ strtolower($id) }}')```

**then i got this error**

```FatalErrorException in QueryBuilderEngine.php line 20:
Class Yajra\Datatables\Engines\QueryBuilderEngine contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Yajra\Datatables\Contracts\DataTableEngineContract::filtering)```

**Comment by alehosaini** [comment]
Thanks for your updates.
I used the new code its took half time of old code.
in other way perfomance problem is still exists.
you can looking on this images how in website code took more than phpMyAdmin execuation time.

![2](https://user-images.githubusercontent.com/16884624/29632027-4ce919ea-884a-11e7-8238-f6792bee148d.png)
![233](https://user-images.githubusercontent.com/16884624/29632026-4ce7d0c6-884a-11e7-84d6-2ab844e397ad.png)

```select contacts.id as id, contacts.level, contacts.firstname, contacts.lastname, CONCAT(contacts.firstname," ",contacts.lastname) as contact_name, (SELECT value FROM book WHERE module = "Contacts" AND type = "Number" AND module_id = contacts.id LIMIT 1) as tel, (SELECT value FROM book WHERE module = "Contacts" AND type = "Email" AND module_id = contacts.id LIMIT 1) as email, (SELECT date FROM activities WHERE type != "Note" AND contact_id = contacts.id ORDER BY id DESC LIMIT 1) as last_relation, company.name as company, company.id as company_id, owner.name as owner_name from `contacts` left join `users` as `owner` on `contacts`.`owner_id` = `owner`.`id` left join `companies` as `company` on `contacts`.`company_id` = `company`.`id` left join `activities` as `activity` on `contacts`.`id` = `activity`.`contact_id` and `activity`.`type` != 'Note' where `contacts`.`client_id` = 1 and `contacts`.`id` < 20 group by `contacts`.`id` order by `id` desc limit 20```

**Comment by alehosaini** [comment]
Thanks the code work fine.

**Comment by yajra** [comment]
@alehosaini thanks for the feedback. PR merged and released on v7.10.0.
